src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -4564,28 +4566,14 @@ def invoke_action(self, action, request_body):
         body_content_kwargs = {}
         body_content_kwargs['content'] = json.loads(request_body) if request_body else None
 
-        # Construct and send request
-        def long_running_send():
-            request = client.post(url, query_parameters, header_parameters, **body_content_kwargs)
-            pipeline_response = client._pipeline.run(request, stream=False)
-            return pipeline_response.http_response.internal_response
-
-        def get_long_running_status(status_link, headers=None):
-            request = client.get(status_link, query_parameters, header_parameters)
-            if headers:
-                request.headers.update(headers)
-            pipeline_response = client._pipeline.run(request, stream=False)
-            return pipeline_response.http_response.internal_response
-
-        def get_long_running_output(response):
-            from azure.core.exceptions import HttpResponseError
-            if response.status_code not in [200, 202, 204]:
-                exp = HttpResponseError(response)
-                exp.request_id = response.headers.get('x-ms-request-id')
-                raise exp
-            return response.text
-
-        return AzureOperationPoller(long_running_send, get_long_running_output, get_long_running_status)
+        def deserialization_cb(pipeline_response):
+            return json.loads(pipeline_response.http_response.text())
+
+        request = client.post(url, query_parameters, header_parameters, **body_content_kwargs)
+        pipeline_response = client._pipeline.run(request, stream=False)
+
+        return LROPoller(client=client, initial_response=pipeline_response, deserialization_callback=deserialization_cb,
+                         polling_method=ARMPolling(lro_options={"final-state-via": "azure-async-operation"}))
```

Comment:
Could we still obtain the long running status after the migration?

Reply:
Yes, it looks the same as before.
![image](https://github.com/user-attachments/assets/bd313ac4-ac48-4634-a86d-3ab2dc2c958d)
![image](https://github.com/user-attachments/assets/4ad60191-e6a2-43e9-952a-791c7f127c3f)